Tools/serial/generate_config.py

```diff
@@ -0,0 +1,301 @@
+#! /usr/bin/env python
+""" Script to generate Serial (UART) parameters and the ROMFS startup script """
+
+from __future__ import print_function
+
+import argparse
+import os
+import sys
+
+from jinja2 import Environment, FileSystemLoader
+
+try:
+    import yaml
+except:
+    print("Failed to import yaml.")
+    print("You may need to install it with 'sudo pip install pyyaml'")
+    print("")
+    raise
+
+
+## Configuration
+
+# All possible Serial ports
+# Note: do not re-use or change indexes. When adding a port, always use an
+# index that has never been used before. This is important for compatibility
+# with QGC (parameter metadata)
+serial_ports = {
+    # index 0 means disabled
+    "GPS1": {
+        "label": "GPS 1",
+        "index": 1,
+        "default_baudrate": 0,
+        },
+    "GPS2": {
+        "label": "GPS 2",
+        "index": 2,
+        "default_baudrate": 0,
+        },
+    "TEL1": {
+        "label": "TELEM 1",
+        "index": 3,
+        "default_baudrate": 57600,
+        },
+    "TEL2": { # companion port
+        "label": "TELEM 2",
+        "index": 4,
+        "default_baudrate": 921600,
+        },
+    "TEL3": {
+        "label": "TELEM 3",
+        "index": 5,
+        "default_baudrate": 57600,
+        },
+    "TEL4": {
+        "label": "TELEM 4",
+        "index": 6,
+        "default_baudrate": 57600,
+        },
+    "URT3": { # for Omnibus
```

Comment:
I think we'll end up needing the full range of generic names (especially for the linux boards).